pkgs/_macro_tool/bin/main.dart

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:_macro_tool/macro_tool.dart';
+import 'package:args/args.dart';
+import 'package:path/path.dart' as p;
+
+final argParser = ArgParser()
+  ..addOption('workspace', help: 'Path to workspace.')
+  ..addOption('packageConfig', help: 'Path to package config.')
+  ..addOption('script', help: 'Path to script.')
+  ..addFlag('skip-cleanup');
+
+Future<void> main(List<String> arguments) async {
+  final args = argParser.parse(arguments);
+  final workspace = args['workspace'] as String?;
+  final packageConfig = args['packageConfig'] as String?;
+  final script = args['script'] as String?;
+
+  if (workspace == null || packageConfig == null || script == null) {
```

Comment:
Can we not just default at least the workspace and packageConfig args? (to `./` and regular package config discovery based on the script path?).

Reply:
I'd prefer to keep it simple for now, explicit should work nicely for e2e tests.

If we end up actually running it by hand we can make it smarter :)